pkg/work/work.go

```diff
@@ -0,0 +1,41 @@
+package work
+
+import "sync"
+
+// Worker must be implemented by types that want to use the worker pool.
+type Worker interface {
+	Task()
+}
+
+// Pool provides a pool of goroutines that can execute any Worker tasks
+// that are submitted
+type Pool struct {
+	tasks chan Worker
+	wg    sync.WaitGroup
+}
+
+func New(maxGoroutines int) *Pool {
+	p := Pool{
+		tasks: make(chan Worker),
+	}
+	p.wg.Add(maxGoroutines)
+	for i := 0; i < maxGoroutines; i++ {
```

Comment:
I know the current code is much simpler and clearer for supporting task queue. However, it seems there is an obvious drawback that there will be always maxGoroutines goroutines created and pending on the task chan even there are not so many requests.

Maybe an infinite `for` with `select` way is much better?

Reply:
TBH. I don't have strong opinions on this one. For me having this kind of limit is beneficial to control the workload in case of DDoS or spike in number of scan requests. On the other hand I don't think the overhead of creating idle routines is so big.

I've added metrics endpoint so we can have exact measures and if memory will be concerned, we can always improve it.